toolchain/check/testdata/array/init_dependent_bound.carbon

```diff
@@ -14,7 +14,7 @@
 
 library "[[@TEST_NAME]]";
 
-fn G(T:! type) {
+fn G(T:! Core.Destroy) {
   // We can initialize this without knowing T.
   //@dump-sem-ir-begin
   var arr: array(T, 0) = ();
```

Comment:
I wonder if arrays with bound `0` should be a special case and not require `T` to be destroyable here?